### ISSUE TITLE
🐛 Bug Report: 400 Error Even Though Index Exists

### ISSUE THREAD

**Issue by stnguyen90** [appwrite/appwrite#2716]
### 👟 Reproduction steps

1. Create a collection with 2 attributes: `firstName`, `lastName`
2. Create an index with those two attributes like `firstName`, `lastName`
3. Make an API call with 2 queries: `lastName.equal("a")`, `firstName.equal("b")`

### 👍 Expected behavior

The API call returns status 200

### 👎 Actual Behavior

The API call returns status 400 saying an index for `lastName`, `firstName` is missing.

### 🎲 Appwrite version

Version 0.12.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
It seems like the problem is in the [Queries Validator](https://github.com/utopia-php/database/blob/ecc143f2cfe16b23675407035c6b5375ba263285/src/Database/Validator/Queries.php#L171) where it checks the queries against the index. 

```php
    protected function arrayMatch($indexes, $queries): bool
    {
        // Check the count of indexes first for performance
        if (count($indexes) !== count($queries)) {
            return false;
        }

        // Only matching arrays will have equal diffs in both directions
        if (array_diff_assoc($indexes, $queries) !== array_diff_assoc($queries, $indexes)) {
            return false;
        }

        return true;
    }
```

`array_diff_assoc()` checks the order too. Perhaps `array_diff()` would be better.

**Comment by stnguyen90** [comment]
Related PR https://github.com/utopia-php/database/pull/151

**Comment by TorstenDittmann** [comment]
This got implemented and released with Appwrite 1.0.0 🥳 

Thank you for your contribution 🙏🏻  You can read more [here](https://appwrite.io/1.0).
